frontend/public/components/monitoring/alert-manager-config.tsx

```diff
@@ -74,36 +82,119 @@ const ReceiverTableHeader = () => {
   ];
 };
 ReceiverTableHeader.displayName = 'ReceiverTableHeader';
 
-const getIntegrationTypes = (receiver) => {
+const getIntegrationTypes = (receiver: AlertManagerReceiver): string[] => {
   /* Given receiver = {
        "name": "team-X-pager",
        "email_configs": [...],
        "pagerduty_configs": [...]
      };
-     returns "email, pagerduty"
+     returns ['email', 'pagerduty']
   */
   const integrationTypes = _.filter(_.keys(receiver), (key) => _.includes(key, '_configs'));
-  return _.join(_.map(integrationTypes, (type) => type.substr(0, type.indexOf('_configs'))), ', ');
+  return _.map(integrationTypes, (type) => type.substr(0, type.indexOf('_configs')));
 };
 
-// Recursive function to get hierarchy of routing labels for each receiver
-const getRoutingLabels = (routes, parentLabels) => {
-  let results = [];
+/**
+ * Recursive function which transverses routes and sub-routes to get labels for each receiver.
+ * Each entry is a set of labels used to route alerts to a receiver
+ *
+ * Ex: returns
+ * [{
+ *   "receiver": "team-Y-pager",
+ *   "labels": {
+ *     "service": "database",
+ *     "owner": "team-Y"
+ *   }
+ * },
+ * {
+ *   "receiver": "team-Y-pager",
+ *   "labels": {
+ *     "service": "files",
+ *     "severity": "critical"
+ *   }
+ * }]
+}*/
+const getRoutingLabelsByReceivers = (routes, parentLabels): RoutingLabelsByReceivers[] => {
+  let results: RoutingLabelsByReceivers[] = [];
   let labels = {};
   for (const obj of routes) {
     labels = _.merge({}, parentLabels, obj.match, obj.match_re);
     results.push({ receiver: obj.receiver, labels });
     if (obj.routes) {
-      results = results.concat(getRoutingLabels(obj.routes, labels));
+      results = results.concat(getRoutingLabelsByReceivers(obj.routes, labels));
     }
   }
   return results;
 };
 
+/**
+ * Is receiver used in a top-level route that has no sub routes, or
+ * is receiver not in any route (no routing labels)?
+ */
+const hasSimpleRoute = (
+  receiver: AlertManagerReceiver,
+  receiverRoutingLabels: RoutingLabelsByReceivers[],
+): boolean => {
+  const routes = _.get(config, ['route', 'routes']);
+  return (
+    _.filter(routes, (route) => {
+      return route.receiver === receiver.name && _.isUndefined(route.routes);
+    }).length > 0 || _.isEmpty(receiverRoutingLabels)
+  );
+};
+
+/**
+ * Does receiver contains a single known receiver type (ex: pagerduty), which has a single config.
+ * No receiver type specified is valid, as well as a single receiver type with no config
+ */
+const hasSimpleReceiver = (receiver: AlertManagerReceiver, integrationTypes: string[]): boolean => {
+  if (integrationTypes.length === 0) {
+    return true;
+  }
+
+  const singleKnownIntegrationType =
+    integrationTypes.length === 1 && isKnownReceiverType(integrationTypes[0]);
+
+  if (!singleKnownIntegrationType) {
+    return false;
+  }
+
+  const numConfigs = _.get(receiver, `${integrationTypes[0]}_configs`).length;
+  return numConfigs <= 1;
+};
```

Comment:
I think this is equivalent to

```
const hasSimpleReceiver = (receiver: AlertManagerReceiver): boolean => {
  const knownReceiverTypes = ['pagerduty_configs', 'webhook_configs'];
  const simpleReceivers = knownReceiverTypes.filter(
    (type) => receiver[type] && receiver[type].length <= 1,
  );
  return simpleReceivers.length <= 1;
};
```

Then you don't need `receiverTypes` or `isKnownReceiverType`.

Reply:
Umm...logic didn't hold up, especially for an unknown type such as 'webhook_configs'.  Updated the code though to be more straight forward.